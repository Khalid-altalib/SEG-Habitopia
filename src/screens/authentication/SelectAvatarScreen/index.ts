import SelectAvatarScreen from "./SelectAvatarScreen";

export default SelectAvatarScreen;
