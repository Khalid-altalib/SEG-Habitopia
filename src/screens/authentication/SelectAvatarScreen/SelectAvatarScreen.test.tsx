// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Redux
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

// Habitopia
import SelectAvatarScreen from "@screens/authentication/SelectAvatarScreen";
import TestingWrapper from "@app/testingWrapper";

it("Renders correctly.", () => {
  const mockStore = configureStore([thunk])({});

  render(
    <TestingWrapper store={mockStore}>
      <SelectAvatarScreen />
    </TestingWrapper>
  );

  expect(screen.toJSON()).toMatchSnapshot();
});
