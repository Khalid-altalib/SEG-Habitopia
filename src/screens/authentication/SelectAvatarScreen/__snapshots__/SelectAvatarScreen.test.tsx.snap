// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4293190887,
          4294243573,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      {
        "height": "100%",
      }
    }
  >
    <View
      style={
        {
          "height": "100%",
          "padding": 25,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "height": "100%",
            "paddingTop": "10%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#18181b",
              },
              {
                "fontFamily": undefined,
                "fontSize": 40,
              },
              undefined,
            ]
          }
        >
          Pick your avatar
        </Text>
        <Text
          style={
            [
              {
                "color": "#18181b",
              },
              {
                "color": "#71717a",
                "fontFamily": undefined,
                "fontSize": 15,
              },
              {
                "marginBottom": "10%",
                "marginTop": 5,
              },
            ]
          }
        >
          You will be able to change it later.
        </Text>
        <View
          isFullWidth={true}
          style={
            {
              "marginTop": 20,
            }
          }
          type={1}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                {
                  "colors": [
                    4288782762,
                    4285624698,
                  ],
                  "endPoint": [
                    1,
                    1,
                  ],
                  "locations": undefined,
                  "startPoint": [
                    0,
                    0,
                  ],
                }
              }
              style={
                [
                  {
                    "alignItems": "center",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingHorizontal": 40,
                    "paddingVertical": 15,
                  },
                  {
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#18181b",
                    },
                    {
                      "color": "#e4e4e7",
                      "fontFamily": undefined,
                      "fontSize": 19,
                    },
                    [
                      null,
                      {
                        "alignSelf": "center",
                      },
                    ],
                  ]
                }
              >
                Camera
              </Text>
              <View
                style={
                  {
                    "marginLeft": 5,
                  }
                }
              />
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
        <View
          isFullWidth={true}
          style={
            {
              "marginTop": 20,
            }
          }
          type={1}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                {
                  "colors": [
                    4288782762,
                    4285624698,
                  ],
                  "endPoint": [
                    1,
                    1,
                  ],
                  "locations": undefined,
                  "startPoint": [
                    0,
                    0,
                  ],
                }
              }
              style={
                [
                  {
                    "alignItems": "center",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingHorizontal": 40,
                    "paddingVertical": 15,
                  },
                  {
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#18181b",
                    },
                    {
                      "color": "#e4e4e7",
                      "fontFamily": undefined,
                      "fontSize": 19,
                    },
                    [
                      null,
                      {
                        "alignSelf": "center",
                      },
                    ],
                  ]
                }
              >
                Gallery
              </Text>
              <View
                style={
                  {
                    "marginLeft": 5,
                  }
                }
              />
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
        <View
          isFullWidth={true}
          onPress={[Function]}
          style={
            {
              "marginTop": 20,
            }
          }
          type={1}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginTop": 20,
                "opacity": 1,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                {
                  "colors": [
                    4288782762,
                    4285624698,
                  ],
                  "endPoint": [
                    1,
                    1,
                  ],
                  "locations": undefined,
                  "startPoint": [
                    0,
                    0,
                  ],
                }
              }
              style={
                [
                  {
                    "alignItems": "center",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingHorizontal": 40,
                    "paddingVertical": 15,
                  },
                  {
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#18181b",
                    },
                    {
                      "color": "#e4e4e7",
                      "fontFamily": undefined,
                      "fontSize": 19,
                    },
                    [
                      null,
                      {
                        "alignSelf": "center",
                      },
                    ],
                  ]
                }
              >
                Instagram
              </Text>
              <View
                style={
                  {
                    "marginLeft": 5,
                  }
                }
              />
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</RNCSafeAreaProvider>
`;
